packages/sanity/src/core/store/bundles/useBundles.ts

```diff
@@ -0,0 +1,26 @@
+import {useMemo} from 'react'
+import {useObservable} from 'react-rx'
+import {useBundlesStore} from 'sanity'
+
+import {type bundlesReducerAction} from './reducer'
+import {type BundleDocument} from './types'
+
+interface BundlesState {
+  data: BundleDocument[] | null
+  error?: Error
+  loading: boolean
+  dispatch: React.Dispatch<bundlesReducerAction>
+}
+
+export function useBundles(): BundlesState {
+  const {state$, dispatch} = useBundlesStore()
+  const state = useObservable(state$)!
+  const bundlesAsArray = useMemo(() => Array.from(state.bundles.values()), [state.bundles])
```

Comment:
Given we now have all of this in rxjs observable and pipe
What do you think if we do this transformation in the observer when it emits?

We could update https://github.com/sanity-io/sanity/pull/7077/files#diff-0972633c3ddf87121161cee5390c665b6e88e1318ba04827eca1dd6ca54ac0fdR223-R227 to:

```ts
  const state$ = merge(listFetch$, listener$, dispatch$).pipe(
    filter((action): action is bundlesReducerAction => typeof action !== 'undefined'),
    scan((state, action) => bundlesReducer(state, action), INITIAL_STATE),
    startWith(INITIAL_STATE),
    map((state) => ({
      ...state,
      bundles: Array.from(state.bundles.values()),
    })),
    shareReplay(1),
  )
```

Reply:
I agree it would be neater, I just didn't want to move around _too much_ that wasn't strictly necessary here.

One thing that gives me pause here is that the bundles array is derived state. Does it make sense to store this duplicate/derived state in the store itself?

One other tiny concern is that it might cause more work and renders on every state change. With the `useMemo` approach, I think changes to things like the loading state would preserve the identity of the `bundles` object and prevent the array being recalculated, whereas the change suggested would cause it to be recalculated and produce a new array on every state change.

Not to say we shouldn't do something like this, I'm just not 100% sure what the best approach is.

Reply:
Makes sense, I didn't considered that other changes will emit a new array causing re renders, you are right, let's keep the change as is.